src/api/app/models/user.rb

```diff
@@ -865,6 +865,16 @@ def send_metric_for_beta_change
     RabbitmqBus.send_to_bus('metrics', "user.#{channel} value=1")
   end
 
+  def run_as
```

Comment:
Nice idea to `yield` but this needs to be unit tested.

Reply:
I added some, if you have other cases in mind that need testing, let me know